crates/extrinsics/src/env_check.rs

```diff
@@ -0,0 +1,384 @@
+use contract_build::Verbosity;
+use ink_metadata::InkProject;
+use scale_info::{
+    form::PortableForm,
+    Field,
+    PortableRegistry,
+    TypeDef,
+};
+
+use anyhow::{
+    Context,
+    Result,
+};
+
+fn get_node_env_fields(
+    registry: &PortableRegistry,
+) -> Result<Option<Vec<Field<PortableForm>>>> {
+    let Some(env_type) = registry.types.iter().find(|t| {
+        let len = t.ty.path.segments.len();
+        t.ty.path.segments[len - 2..] == ["pallet_contracts", "Environment"]
+    }) else {
+        // if we can't find the type, then we use the old contract version.
+        contract_build::verbose_eprintln!(Verbosity::Verbose,
```

Comment:
if you're not passing in the verbosity you might just as well use `eprintln` directly

```suggestion
        eprintln!(
```